### PR TITLE
test(spec): implement sync spec tests.

### LINKED ISSUE #461: Implement `sync` spec test runner.
This one is is the same as the fork choice one

https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/sync

### PR DESCRIPTION
**Pull request by mpaulucci**
**Description**
Implements sync test runner.

Closes #461

### CODE REVIEW

File: lib/lambda_ethereum_consensus/execution/engine_api/mocked.ex [hunk 1]
```diff
@@ -20,6 +20,6 @@ defmodule LambdaEthereumConsensus.Execution.EngineApi.Mocked do
 
   @spec forkchoice_updated(map, map | any) :: {:ok, any} | {:error, any}
   def forkchoice_updated(_forkchoice_state, _payload_attributes) do
-    {:ok, %{"payload_id" => nil, payload_status: %{"status" => "SYNCING"}}}
+    {:ok, %{"payload_id" => nil, "payload_status" => %{"status" => "SYNCING"}}}
```
Comment (MegaRedHand) [inline lib/lambda_ethereum_consensus/execution/engine_api/mocked.ex:23]: Shouldn't we return this already parsed?

File: lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex [hunk 2]
```diff
@@ -181,20 +180,10 @@ defmodule LambdaEthereumConsensus.ForkChoice do
   def recompute_head(store) do
     {:ok, head_root} = Helpers.get_head(store)
 
-    head_block = Blocks.get_block!(head_root)
-    head_execution_hash = head_block.body.execution_payload.block_hash
+    Handlers.notify_forkchoice_update(store, head_root)
 
+    head_block = Blocks.get_block!(head_root)
```
Comment (MegaRedHand) [inline lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:185]: We should also pass the `head_block`, to avoid fetching the same block twice

Comment (mpaulucci) [reply lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:185]: Good call!

File: lib/lambda_ethereum_consensus/execution/engine_api/mocked.ex (hunk 1 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/execution/engine_api/mocked.ex:23]: Yes, that's what I mean. It seems I mistook the `EngineApi` and `ExecutionClient` modules 😭

**Review by MegaRedHand — APPROVED** [review]
Nice work! 🚀

**Review by MegaRedHand — APPROVED** [review]

**Pull request merged**